consensus/src/dag/dag_handler.rs

```diff
@@ -37,22 +38,40 @@ impl NetworkHandler {
         signer: ValidatorSigner,
         epoch_state: Arc<EpochState>,
         storage: Arc<dyn DAGStorage>,
+        payload_client: Arc<dyn PayloadClient>,
+        network_sender: Arc<dyn DAGNetworkSender>,
+        time_service: Arc<dyn TimeService>,
     ) -> Self {
+        let rb = Arc::new(ReliableBroadcast::new(
+            epoch_state.verifier.get_ordered_account_addresses().clone(),
+            network_sender,
+        ));
         Self {
             dag_rpc_rx,
             node_receiver: NodeBroadcastHandler::new(
                 dag.clone(),
-                signer,
+                signer.clone(),
                 epoch_state.clone(),
+                storage.clone(),
+            ),
+            dag_driver: DagDriver::new(
+                signer.author(),
+                epoch_state.clone(),
+                dag.clone(),
+                payload_client,
+                rb,
+                1,
```

Comment:
Why cannot you hard code this? Why would anyone call it with a round different than 1?
Moreover, you handle the 0 case, but what if someone calls it with 17? this can cause issues, right?

Reply:
We can't start from 1 at all times because 1..16 can be GCed in all nodes and we can only start from round 17. This will change as we add state-sync logic and integrate to epoch manager.

Reply:
I see, good point. But in this case you should not check parents for nodes in round 17.

Reply:
Yes, I will add a TODO